evap/staff/templates/staff_evaluation_operation.html

```diff
@@ -55,9 +56,39 @@
         {% endif %}
         <input type="hidden" name="target_state" value="{{ target_state }}" />
 
+        {% if show_delete_answers_checkbox %}
+            <div class="card mb-3">
+                <div class="card-body">
+                    <div class="row">
+                        <div class="col">
+                            <div class="form-check">
+                                <input class="form-check-input" type="checkbox" name="delete-previous-answers"
+                                       id="delete-previous-answers" checked>
+                                <label class="form-check-label" for="delete-previous-answers">
+                                    {% trans "Delete previous answers" %}
+                                </label>
+                            </div>
+                        </div>
+                    </div>
+                </div>
+            </div>
+        {% endif %}
+
         <div class="card card-submit-area card-submit-area-2 text-center mb-3">
             <div class="card-body">
-                <button type="submit" class="btn btn-primary">{% translate 'Confirm' %}</button>
+                <!-- TODO: only show modal if show_delete_answers_checkbox and #delete-previous-answers is set -->
+                <confirmation-modal type="submit" form="evaluation-operation-form">
+                    <span slot="title">{% translate 'Reset Evaluation' %}</span>
+                    <span slot="action-text">{% translate 'Reset Evaluation' %}</span>
+                    <!-- TODO: Brainstorm better modal-message -->
+                    <span slot="question">
+                        {% blocktranslate trimmed %}
+                            This will delete all previously received answers.
+                        {% endblocktranslate %}
+                    </span>
+                    <!-- TODO: when should one use type="button" vs type="submit" while using the confirmation-model dialogue? -->
+                    <button slot="show-button" type="button" class="btn btn-primary">{% translate 'Confirm' %}</button>
+                </confirmation-modal>
```

Comment:
@niklasmohrin As I understand it, you worked on the new `<confirmation-modal>`. Do you have an idea for a nice way to skip the modal or only show it when certain conditions are met?

Reply:
The modal display is triggered here: https://github.com/e-valuation/EvaP/blob/4d998159bff118e5ac14529201cc8da9695e1e80/evap/static/ts/src/confirmation-modal.ts#L30

You could make this single line available as a method on `ConfirmationModal` and then use JavaScript to dynamically use the modal or `form.requestSubmit()` directly. Your JavaScript can then use an arbitrary condition for when to use which of the two.

Does this help?

Reply:
I added the option to set `skipShowingModal`, a generic `()=>boolean`-function, on the modal. With this one can easily specify the logic next to the modal in question:

```html
<script>
    const modal = document.getElementById("confirmation-modal");
                                                                           
    function isDeleteAnswersUnchecked() {
        return !document.getElementById("delete-previous-answers").checked;
    }
                                                                           
    modal.skipShowingModal = isDeleteAnswersUnchecked;
</script>
```

What do you think?

Reply:
After thinking about it again, maybe it would be better if "submit without modal" would not be the default. Instead, the extra button could just be ˋtype=buttonˋ and either trigger the modal, or do ˋform.requestSubmit() otherwise